libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1142,19 +1142,106 @@ genFromSpecT (simplifySpec -> spec) = case spec of
 
 shrinkWithSpec :: forall fn a. HasSpec fn a => Specification fn a -> a -> [a]
 -- TODO: possibly allow for ignoring the `conformsToSpec` check in the `TypeSpec`
--- case when you know what you're doing
+-- case when you know what you're doing (remember to filter on cant)
 shrinkWithSpec (simplifySpec -> spec) a = filter (`conformsToSpec` spec) $ case spec of
-  -- TODO: filter on can't if we have a known to be sound shrinker
   TypeSpec s _ -> shrinkWithTypeSpec @fn s a
-  -- TODO: The better way of doing this is to compute the dependency graph,
-  -- shrink one variable at a time, and fixup the rest of the variables
-  SuspendedSpec {} -> shr a
-  MemberSpec {} -> shr a
+  SuspendedSpec x p -> shr a ++ shrinkFromPreds p x a
+  MemberSpec {} -> shr a -- TODO: with a defined shrink order we would be able to use the elements here
```

Comment:
I know this is only a comment, but what do you mean by a defined order, and how would that help?